src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -308,7 +308,18 @@ private static Instrumenter RemoveDynamicAnalysisInjectors(Instrumenter instrume
                     return RemoveDynamicAnalysisInjectors(previous);
                 case DebugInfoInjector { Previous: var previous } injector:
                     var newPrevious = RemoveDynamicAnalysisInjectors(previous);
-                    return (object)newPrevious != previous ? new DebugInfoInjector(newPrevious) : injector;
+                    if ((object)newPrevious == Instrumenter.NoOp)
```

Comment:
>(object)newPrevious == Instrumenter.NoOp [](start = 24, length = 40)

I think it would be better to check if ```(object)newPrevious == previous``` first